client/src/frontend/components/resource-slide/Slide/index.js

```diff
@@ -0,0 +1,55 @@
+export * from "./Info";
+export Zoom from "./Zoom";
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Zoom from "./Zoom";
+import { Info } from "./Info";
+
+export default class ResourceListSlide extends Component {
+  static propTypes = {
+    resource: PropTypes.object.isRequired,
+    zoom: PropTypes.bool,
+    zoomLabel: PropTypes.string,
+    info: PropTypes.bool,
+    backgroundUrl: PropTypes.string
+  };
+
+  static defaultProps = {
+    zoom: false,
+    zoomLabel: "Zoom",
+    info: true
+  };
+
+  render() {
+    const {
+      resource,
+      zoom,
+      zoomLabel,
+      children,
+      info,
+      backgroundUrl
+    } = this.props;
+
+    if (!resource) return null;
+
+    const bgImage = backgroundUrl || "/static/images/resource-splash.png";
+
+    let slideStyles = {};
+    if (React.Children.count(children) === 0 || !!backgroundUrl) {
+      slideStyles = { backgroundImage: `url(${bgImage})` };
+    }
```

Comment:
Too much logic in the render method that's too hard to easily parse. You have a tendency to do this—just move small bits into class instance methods. The act of naming those methods really makes the code easier for the next person to read. For example:

```
get background() {
  return this.props.backgroundUrl || "/static/images/resource-splash.png"
}

get hasBackgroundUrl() {
  return !!this.props.backgroundUrl;
}

get hasChildren() {
  return React.Children.count(children) === 0
}

get hasBackground() {
  if (this.hasChildren) return true;
  if (this.hasBackgroundUrl) return true;
  return false;
}

render() {
  [...]
  const slideStyles = this.hasBackground ? { backgroundImage: url(this.background) } : {} 
}
```

The difference between this and what you wrote is really important to me—my version might be more verbose, but whoever comes to this next can understand the logic without having to really grok the code. Note, too, that when we use method names to explain what's happening here, we see that the logic doesn't actually make much sense. If I read your code right, `this.background` would always have a value (either the provided value or the fallback in /static), yet you're only showing a background if the provided value is present. It seems to me that the logic is wrong, and it should be simplified to this:

```
get background() {
  return this.props.backgroundUrl || "/static/images/resource-splash.png"
}

get hasChildren() {
  return React.Children.count(children) === 0
}

get hasBackground() {
  return this.hasChildren;
}

render() {
  [...]
  const slideStyles = this.hasBackground ? { backgroundImage: url(this.background) } : {} 
}
```